QAOA/src/QaoaHybrid/HybridQaoa.cs

```diff
@@ -0,0 +1,261 @@
+﻿namespace Microsoft.Quantum.QAOA.QaoaHybrid
+
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using Accord.Math.Optimization;
+    using Microsoft.Quantum.Simulation.Core;
+    using Microsoft.Quantum.Simulation.Simulators;
+
+    //currently support up to 2-local Hamiltonians; will be generalized later
+    public class HybridQaoa 
+    {
+        private Utils.FreeParamsVector freeParamsVector;
+        private readonly int numberOfIterations;
+        private readonly int p;
+        private readonly ProblemInstance problemInstance;
+        private double bestHamiltonianValue;
+        private string bestVector;
+        private double[] bestBeta;
+        private double[] bestGamma;
+        private readonly int numberOfRandomStartingPoints;
+        private readonly QaoaLogger logger;
+        private readonly bool shouldLog;
+
+        public HybridQaoa(int numberOfIterations, int p, ProblemInstance problemInstance, int numberOfRandomStartingPoints = 1, bool shouldLog = false, double[] initialBeta = null, double[] initialGamma = null)
+        {
+
+            this.numberOfIterations = numberOfIterations;
+            this.p = p;
+            this.problemInstance = problemInstance;
+            this.freeParamsVector.beta = initialBeta;
+            this.freeParamsVector.gamma = initialGamma;
+            this.bestHamiltonianValue = double.MaxValue;
+            this.bestVector = null;
+            this.numberOfRandomStartingPoints = numberOfRandomStartingPoints;
+            this.shouldLog = shouldLog;
+            if (shouldLog)
+            {
+                this.logger = new QaoaLogger();
+            }
+        }
+
+
+        /// # Summary
+        /// Calculates the value of the cost function based on costs provided.
+        ///
+        /// # Input
+        /// ## result
+        /// A binary string. In this context it is a result that we get after measuring the QAOA state.
+        ///
+        /// ## costs
+        /// A list of costs for the cost function.
+        /// 
+        /// # Output
+        /// The value of the costfunction.
```

Comment:
It looks like these documentation comments use Q# formatting (Markdown) rather than C# formatting (XML)?
```suggestion
        /// <summary>
        ///     Calculates the value of the cost function based on costs provided.
        /// </summary>
        /// <param name="result">
        ///     A binary string. In this context it is a result that we get after measuring the QAOA state.
        /// </param>
        /// <param name="costs">
        ///     A list of costs for the cost function.
        /// </param>
        /// <returns>
        ///     The value of the cost function.
        /// </returns>
```